src/irr/asset/CPLYMeshFileLoader.cpp

```diff
@@ -19,15 +19,81 @@ namespace irr
 namespace asset
 {
 
-// constructor
-CPLYMeshFileLoader::CPLYMeshFileLoader()
+_IRR_STATIC_INLINE_CONSTEXPR const char* VERT_SHADER_CACHE_KEY = "irr/builtin/shader/loaders/ply/vertex";
+_IRR_STATIC_INLINE_CONSTEXPR const char* FRAG_SHADER_CACHE_KEY = "irr/builtin/shader/loaders/ply/fragment";
+
+const char* VERT_SHADER = R"===(
+#version 430 core
+layout(location = 0) in vec4 vPos;
+layout(location = 1) in vec4 vCol;
+layout(location = 3) in vec3 vNormal; 
+
+#include <irr/builtin/glsl/vertex_utils/vertex_utils.glsl>
+
+layout (set = 1, binding = 0, row_major, std140) uniform UBO {
+    irr_glsl_SBasicViewParameters params;
+} CamData;
+
+layout (location = 0) out vec3 color;
+layout (location = 1) out vec3 viewPos;
+layout (location = 2) out vec3 normal;
+
+void main()
+{
+    gl_Position = irr_glsl_pseudoMul4x4with3x1(irr_builtin_glsl_workaround_AMD_broken_row_major_qualifier_mat4x4(CamData.params.MVP), vPos);
+	viewPos = irr_glsl_pseudoMul3x4with3x1(irr_builtin_glsl_workaround_AMD_broken_row_major_qualifier_mat4x3(CamData.params.MV), vPos);
+	mat3 normalMat = irr_glsl_SBasicViewParameters_GetNormalMat(CamData.params);
+    normal = normalMat*normalize(vNormal);
+    color = vCol.xyz;
+}
+)===";
+
+const char* FRAG_SHADER = R"===(
+#version 430 core
+
+layout (location = 0) in vec3 color;
+layout (location = 1) in vec3 viewPos;
+layout (location = 2) in vec3 normal;
+
+layout(location = 0) out vec4 pixelColor;
+
+void main()
 {
+	// blinn-phong maybe?
```

Comment:
nah, just vertex color multiplied by a "fake" (procedural) texture generated from UV coordinates (think checkerboard)

Reply:
so VCol attribute times texture() color in that case, but what when there is no UV attribute? Should I use uber shader then and consider lack of that attribute?

Reply:
okay, so I will use dummy texture from buildin Assets, and only pipeline layout left to create there